src/jupyter/assignments.unit.test.ts

```diff
@@ -477,6 +482,112 @@ describe("AssignmentManager", () => {
     });
   });
 
+  describe("getRemoteServers", () => {
+    const endpointWithName = "test-endpoint-with-name";
+    const endpointWithoutName = "test-endpoint-without-name";
+    const endpointWithoutSession = "test-endpoint-without-session";
+    const allAssignments = [
+      {
+        endpoint: endpointWithName,
+        variant: Variant.DEFAULT,
+        machineShape: Shape.STANDARD,
+        accelerator: "",
+      },
+      {
+        endpoint: endpointWithoutName,
+        variant: Variant.DEFAULT,
+        machineShape: Shape.STANDARD,
+        accelerator: "",
+      },
+      {
+        endpoint: endpointWithoutSession,
+        variant: Variant.DEFAULT,
+        machineShape: Shape.STANDARD,
+        accelerator: "",
+      },
+    ];
+    const defaultSession = {
+      id: "",
+      path: "",
+      type: "",
+      kernel: {
+        lastActivity: "",
+        executionState: "",
+        id: "",
+        name: "",
+        connections: 1,
+      },
+    };
+
+    beforeEach(() => {
+      colabClientStub.listAssignments.resolves(allAssignments);
+      colabClientStub.listSessions.withArgs(endpointWithName).resolves([
+        {
+          ...defaultSession,
+          name: "test-session-name",
+        },
+      ]);
+      colabClientStub.listSessions.withArgs(endpointWithoutName).resolves([
+        {
+          ...defaultSession,
+          name: "",
+        },
+      ]);
+      colabClientStub.listSessions
+        .withArgs(endpointWithoutSession)
+        .resolves([]);
```

Comment:
Rather than define all of the test variants in the `describe` and setup each of their matching stubs in a `beforeEach`, could we instead have the test tell the full story? It's okay if in some we need to do "duplicate" setup, to keep the tests fully descriptive (internal guidance go/tott/643).

Right now just reading the test body, it doesn't tell the full story, which forces readers to bounce between the setup blocks here and the test code.

Reply:
Re-opening, mind addressing?